consensus/src/experimental/commit_phase.rs

```diff
@@ -190,73 +144,52 @@ impl CommitPhase {
         &mut self,
         commit_decision: CommitDecision,
     ) -> anyhow::Result<()> {
-        let li = commit_decision.ledger_info();
+        if let Some((vecblock, leder_info)) = self.blocks.as_ref() {
+            let li = commit_decision.ledger_info();
 
-        if li.ledger_info().commit_info().round() < self.committed_round {
-            return Ok(()); // we ignore the message
+            // if the block infos do not match
+            if !(li.ledger_info().commit_info().clone()
+                == leder_info.ledger_info().commit_info().clone())
+            {
+                return Ok(()); // ignore the message
+            }
+
+            commit_decision.verify(&self.verifier)?;
+
+            // replace the signature tree
+            self.blocks = Some((vecblock.clone(), li.clone()));
         }
 
-        commit_decision.verify(&self.verifier)?;
+        Ok(())
+    }
 
-        let executed_state_hash = li.ledger_info().commit_info().executed_state_id();
+    pub async fn process_block(&mut self) -> anyhow::Result<()> {
```

Comment:
the name is misleading, I thought this is to process the blocks sent by execution, maybe it's better to call it check_commit

Reply:
resolved